code/script.js

```diff
@@ -244,100 +246,115 @@ const generateBoard = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
-// This function to start (and restart) the game
+// Starts the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard()
+  setSecret()
+  winOrLose.style.display = 'none'
+
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// This is triggered by a change in the drop-down, and saves the selection as an object "currentQuestion"
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-
+  const value = questions.options[questions.selectedIndex].value
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: value,
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: 'eyeColor',
+      value: value,
+      category: category,
+    }
+    
+  } else if (category === 'accessories' || category === 'other') {
+    currentQuestion = {
+      attribute: value,
+      value: true, 
       category: category,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// Check if the guess is correct or not
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  if (secret[currentQuestion.attribute] === currentQuestion.value) {
+    filterCharacters(true)
+  } else {
+    filterCharacters(false)
+  }
 }
 
-// It'll filter the characters array and redraw the game board.
+
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
-    }
-  } else if (group === 'other') {
-    // Similar to the one above
+  // Alerts...
+  if (currentQuestion.category === 'accessories') {
+    keep ?
```

Comment:
Maybe it would be easier to read the code if you would have continued with if-else statements instead of ternary operators here?

Reply:
or use `switch`?